builtin/providers/fastly/resource_fastly_service_v1.go

```diff
@@ -369,6 +474,60 @@ func resourceServiceV1Update(d *schema.ResourceData, meta interface{}) error {
 			}
 		}
 
+		if d.HasChange("header") {
+			// Note: we don't utilize the PUT endpoint to update a Header, we simply
+			// destroy it and create a new one. This is how Terraform works with nested
+			// sub resources, we only get the full diff not a partial set item diff.
+			// Because this is done on a new version of the configuration, this is
+			// considered safe
```

Comment:
Good note here! I think there are other APIs where we (still) have race conditions because of this behavior. Nice to have it explained out front here.